Update grid blocks preview to match width of products on the frontend

ryelle opened the pull request:
Fixes #488 – from @jobthomas: 

> I'm fairly confident we'll get customers reaching out to us who are frustrated that they can use these blocks full width on the backend, but that it doesn't display frontend.

We can't update the style on the frontend, since the shortcode only renders a 300px image (and we don't want to/can't change all shortcodes). Instead, we can update the preview to match what actually happens on the frontend.

### Screenshots

Here's how some themes handle this 

| Twenty Nineteen | Storefront | Radcliff2 |
| ---------------- | ------------ | --------|
| <img width="655" alt="theme-2019" src="https://user-images.githubusercontent.com/541093/54366854-bbddac00-4647-11e9-95fa-bd30b9d0d779.png"> | <img width="760" alt="theme-storefront" src="https://user-images.githubusercontent.com/541093/54366855-bc764280-4647-11e9-83b7-960f5ffc90c6.png"> | <img width="928" alt="theme-radcliffe" src="https://user-images.githubusercontent.com/541093/54366853-bbddac00-4647-11e9-9ba4-ef57987816aa.png"> |

This PR updates the backend so it always display the single-column as max-width 300px. This number is picked because the 300px image is what's used on the frontend, so most themes restrict the width to 300.

<img width="714" alt="gb-one-col" src="https://user-images.githubusercontent.com/541093/54366955-f34c5880-4647-11e9-88d0-c94347403d57.png">

Two columns + still behaves the same way, though when the content-area is wide it's obvious they still have the 300px max-width:

<img width="938" alt="gb-wide-two-col" src="https://user-images.githubusercontent.com/541093/54367217-68b82900-4648-11e9-9f2b-08478da06cc4.png">

### How to test the changes in this Pull Request:

1. Add a grid block to a post
2. Set the columns down to 1
3. Expect: There should be one item per row, with a max-width of 300px
4. Publish the post, view it
5. Expect: The frontend should match the preview (with the exception of some alignment, as that can be theme dependent)
6. Play with setting columns to 2+, wide or full width, see if it still behaves as expected.

ryelle committed:
Update grid blocks preview to match width of products on the front end

timmyc commented:
> so most themes restrict the width to 300.

I think this is a reasonable approach, but I also feel that this could potentially be a game of css whack-a-mole, as trying to perfect the optimal preview css that fits all theme possibilities could get _interesting_.

Theoretically could theme developers also target the css for the previews to over-ride/update this?

timmyc reviewed (approved):
I had the one question I dropped in the comments, but since this works out-of-the-box for most major/default themes, I think this is a valid approach. LGTM

ryelle commented:
>  as trying to perfect the optimal preview css that fits all theme possibilities could get interesting.

Yeah, you can even see this in my screenshots – Twenty Nineteeen left-aligns the grid, while Storefront and Radcliffe center it. I think it would make the most sense to default to storefront-esque styling, since that's the officially supported "parent theme". 

This restricting to 300px wide actually reflects the image output by the shortcode, so I don't think this part is as much whack-a-mole as it sounds.

> Theoretically could theme developers also target the css for the previews to over-ride/update this?

Yeah, they can add editor styles – I should add some tips about that to the documentation draft 🙂

jameskoster reviewed (approved):
Pre-approving for when the width value respects the core filter for modifying default images sizes.

ryelle committed:
Remove max-width from CSS, set it on the preview itself so we can use the dynamic theme width

ryelle committed:
Override block thumbnail width for multiple column layouts

Previously larger thumbnails would cause multiple column layouts to break and wrap early. This enforces the column width for those cases

ryelle committed:
Add the thumbnail size to images, to prevent larger images in multiple col layouts

ryelle committed:
Update snapshot to new preview markup

ryelle commented:
After some discussion in slack, I've updated the display to use the thumbnail size theme option, to better reflect the size in more themes. This still doesn't match all themes 100%, but at this point we're approaching whack-a-mole territory and themes should be adding editor styles.

(for example, the Hotel theme [changes the width to 622px,](https://cloudup.com/iIdoZNXe5Ro) but then actually displays single-columns in [a side-by-side pattern,](https://cloudup.com/iuyz1PYId4d) so it's not actually 622px wide)

There were a few other fixes needed for larger image sizes, since it can vary now. You can test larger sizes by adding this to a child theme:

```php
function example_store_setup_theme() {
	add_theme_support( 'woocommerce', array(
		'thumbnail_image_width' => 600,
	) );
}
add_action( 'after_setup_theme', 'example_store_setup_theme' );
```

@timmyc or @coderkevin Can I get another dev review?

timmyc reviewed (approved):
Tested multiple columns and did not see any regressions, single column matches preview on the back end LGTM
